### PR TITLE
Plotspc upgrades

### PR DESCRIPTION
**Pull request by dylanbeaudette**
Integrates several ideas from #59

### CODE REVIEW

File: R/plotSPC.R [hunk 1]
```diff
@@ -35,11 +35,254 @@
...
+  col.label=color, 
+  col.palette = rev(brewer.pal(10, 'Spectral')), 
+  col.palette.bias=1, 
+  col.legend.cex=1, 
+  n.legend=8, 
+  lwd=1, 
+  lty=1, 
+  default.color=grey(0.95), 
+  ...
+) {
```
Comment (brownag) [inline R/plotSPC.R:285]: dats nice.

**Comment by brownag** [comment]
Tests generate a warning related to not being able to delete the Rplots.pdf generated by the testing.

Also, this introduces an R CMD check warning. 

```
  Undocumented arguments in documentation object 'plot,SoilProfileCollection,ANY-method'
    'x' 'color' 'width' 'name' 'name.style' 'label' 'hz.depths'
    'alt.label' 'alt.label.col' 'cex.names' 'cex.depth.axis' 'cex.id'
    'font.id' 'print.id' 'id.style' 'plot.order' 'relative.pos' 'add'
    'scaling.factor' 'y.offset' 'x.idx.offset' 'n' 'max.depth'
    'n.depth.ticks' 'shrink' 'shrink.cutoff' 'abbr' 'abbr.cutoff'
    'divide.hz' 'hz.distinctness.offset' 'hz.topography.lty'
    'axis.line.offset' 'plot.depth.axis' 'density' 'col.label'
    'col.palette' 'col.palette.bias' 'col.legend.cex' 'n.legend' 'lwd'
    'lty' 'default.color' '...'
```

**Comment by dylanbeaudette** [comment]
That is really strange: I get no errors. What platform are you on? Likely one that **understands** permissions.

I need to do some research on how to properly document S4 stuff with roxygen. Ideas welcome.

**Comment by brownag** [comment]
No -- this was on the work machine. I pulled the latest version of your branch and ran devtools::test(). Resulted in one warning. 

I agree that it doesnt make sense that I actually didn't have permission. -- it is possible that the plot device got left open.

**Comment by brownag** [comment]
As far as ideas for how to do S4 without creating errors, I can suggest checking out this commit 35b2724 and issue #139

**Comment by brownag** [comment]
These three small commits resolve the S4 errors and uncertainties about the `plot` generic.

https://github.com/ncss-tech/aqp/commit/e8d4bad0daaa80eebdaa1c42d577a3de92cc1908
https://github.com/ncss-tech/aqp/commit/0de044ca264c2dad2ecfd2bc6932daf7a740a0bb
https://github.com/ncss-tech/aqp/commit/0b5ee26597df93d4bf0ba0313d8fe37c2b40764f

**Pull request merged**